### PR TITLE
ensure correct scheme is set for clickhouse repo

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

Possibly fixes https://github.com/plausible/analytics/discussions/1391 by adding `:scheme` option.

More info: https://github.com/plausible/analytics/discussions/1391#discussioncomment-4093013

### Tests
- [x] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -84,6 +84,12 @@ ch_db_url =
     "http://plausible_events_db:8123/plausible_events_db"
   )
 
+ch_db_url_scheme =
+  case URI.parse(ch_db_url) do
+    %URI{scheme: "https"} -> :https
+    _other -> :http
```
Comment (ruslandoga) [inline config/runtime.exs:90]: I'm keeping the prev behaviour this way, by defaulting to `:http`.

**Review by ukutaht — COMMENTED** [review]
Is this how other Ecto libraries work as well?


It feels slightly weird to me that the library itself doesn't use the scheme we provide in the URL. Could this be an issue with clickhouse-ecto or clickhousex instead? If so, I'd prefer to fix it in the library rather than here.

**Comment by ruslandoga** [comment]
Ecto handles the url option itself and passes down only the info extracted from the url. Ecto ignores the scheme. Other adapters are generally not dependant on the scheme and use `:ssl` options instead.

**Comment by ruslandoga** [comment]
Closing as https://github.com/elixir-ecto/ecto/pull/4052 would allow to get the scheme info in the adapter.

**Pull request closed without merging**